### PR TITLE
testing/creduce: upgrade to 2.10.0

### PR DESCRIPTION
**Pull request by J0WI**
Depends on #5639

### CODE REVIEW

**Comment by J0WI** [comment]
Still depending on #5639

**Comment by Ikke** [comment]
This still has issues on some arches:

x86:

> /usr/bin/ld: /usr/bin/../lib/gcc/i586-alpine-linux-musl/8.3.0/../../../libclangFrontend.a(FrontendAction.cpp.o):FrontendAction.cpp:(.text._ZN4llvm13SmallDenseMapINS_16CachedHashStringENS_6detail13DenseSetEmptyELj16ENS_12DenseMapInfoIS1_EENS2_12DenseSetPairIS1_EEE8copyFromERKS8_[_ZN4llvm13SmallDenseMapINS_16CachedHashStringENS_6detail13DenseSetEmptyELj16ENS_12DenseMapInfoIS1_EENS2_12DenseSetPairIS1_EEE8copyFromERKS8_]+0x122): more undefined references to \`__stack_chk_fail_local' follow
/usr/bin/ld: clang_delta: hidden symbol `__stack_chk_fail_local' isn't defined
/usr/bin/ld: final link failed: bad value
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)

armhf:

> clang-6.0: error: unable to execute command: Segmentation fault (core dumped)
clang-6.0: error: clang frontend command failed due to signal (use -v to see invocation)

**Comment by J0WI** [comment]
Rebased on 2.9.0 which also support Clang and LLVM 7.
CI reports the following error on x86:
```
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'. 
```

**Comment by Ikke** [comment]
Relevant part of the build log: http://tpaste.us/RgkY

Not sure if this is related:

>configure:3430: clang -V >&5
>clang-7: error: argument to '-V' is missing (expected 1 value)
>clang-7: error: no input files
>configure:3441: $? = 1
>configure:3430: clang -qversion >&5
>clang-7: error: unknown argument '-qversion', did you mean '--version'?
>clang-7: error: no input files

**Comment by TBK** [comment]
2.10.0 is out https://github.com/csmith-project/creduce/releases/tag/creduce-2.10.0

**Comment by J0WI** [comment]
2.10 depends on LLVM 8

**Comment by Ikke** [comment]
creduce builds fine with gcc under x68, so seems to be an llvm issue.

(tested with a docker i386 image).

File: testing/creduce/APKBUILD [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
+arch="all"
 license="custom"
 # util-linux provides lscpu
 depends="clang indent perl perl-exporter-lite perl-file-which
 	perl-getopt-tabular perl-regexp-common perl-sys-cpu perl-term-readkey
 	util-linux"
 makedepends="clang-dev clang-static flex llvm-dev zlib-dev"
-source="http://embed.cs.utah.edu/creduce/$pkgname-$pkgver.tar.gz"
+source="https://embed.cs.utah.edu/creduce/$pkgname-$pkgver.tar.gz"
 builddir="$srcdir/$pkgname-$pkgver"
```
Comment (tcely) [inline testing/creduce/APKBUILD]:
```suggestion
```

**Comment by tcely** [comment]
You may want to consider switching to `gcc` for `x86` to fix the build problem.

Also, adding a `check` function would be a good improvement.

**Comment by J0WI** [comment]
GCC does not work at all.

**Comment by Ikke** [comment]
@J0WI I could at least build x86 with gcc.

**Comment by J0WI** [comment]
See https://cloud.drone.io/alpinelinux/aports/6813

**Comment by maxice8** [comment]
has conflicts with master

**Comment by J0WI** [comment]
Rebased and updated to 2.10.0

**Comment by maxice8** [comment]
Merged

**Pull request closed without merging**